cunumeric/deferred.py

```diff
@@ -2877,8 +2937,13 @@ def unary_op(
         args: Any,
         multiout: Optional[Any] = None,
     ) -> None:
-        lhs = self.base
-        rhs = src._broadcast(lhs.shape)
+
+        if self.shape == () and self.size == src.size:
+            lhs = self._broadcast(src.shape)
+            rhs = src.base
+        else:
+            lhs = self.base
+            rhs = src._broadcast(lhs.shape)
```

Comment:
I have added this logic for the case when `lhs` has shape `()` and `rhs` has shape `(1,)`. Otherwise we get an error

Reply:
Is this still required? I just ran the test suite without this fix and everything seems to be working.

Reply:
I don't think it is tested. I will add a test